### DISCUSSION TITLE
有计划支持 wav 格式吗

### DISCUSSION THREAD

**Discussion by weilinfox** [the1812/Touhou-Tagger#8] [Ideas]
尝试使用 touhou-tagger 打 tag ，但是似乎并不支持为 wav 格式打 tag ， log 如下：

```shell
$ ls
k3b.log  Track01.wav  Track02.wav  Track03.wav  Track04.wav  Track05.wav  Track06.wav  Track07.wav  Track08.wav  Track09.wav
$ thtag
请输入专辑名称(朱落秋乡): 朱落秋乡
✖ 未找到任何支持的音乐文件.
✖ 未找到任何支持的音乐文件., 进行第1次重试...
✖ 未找到任何支持的音乐文件.
✖ 未找到任何支持的音乐文件., 进行第2次重试...
✖ 未找到任何支持的音乐文件.
✖ 错误: 未找到任何支持的音乐文件.
```

瞅了瞅代码似乎是只支持 mp3 和 flac 格式，想问问有计划支持 wav 格式咩？

**Comment by the1812** [comment]
wav 本来就不支持打 tag 吧

**Reply by weilinfox** [reply]
应该是支持的， [Wikipedia](https://en.wikipedia.org/wiki/WAV#Metadata) 有指出。另外 [Audacity](https://wiki.audacityteam.org/wiki/WAV#Metadata) 、 Kid3 、 mp3tag 都支持为 wav 打 id3 tag 和 RIFF INFO 。

**Reply by the1812** [reply]
噢, 看来只是封面图只有一部分播放器能识别

**Reply by the1812** [reply]
不过 Node.js 这边好像没找到比较合适的库...
(要不还是直接转成 flac 得了

**Reply by weilinfox** [reply]
好像也没找到的样子……有找到[能读的](https://github.com/Borewit/music-metadata)但是没找到能写的。特别是 RIFF INFO 大部分库都不支持写的样子 emmm

**Comment by RingingVolcano** [comment]
想听无损的同时打 tag 最好还是用 flac 吧（
wav 好像问题挺多的

**Comment by weilinfox** [comment]
emmm 找到 python 那边有个 [mutagen](https://github.com/quodlibet/mutagen) 可以用，就[试图实现](https://github.com/weilinfox/haku-thtagger/blob/master/thtagger/models/tagEditor.py#L222-L244)了一下自己的想法：picard、kid3都用 windows1252 或者 utf-8 存的 RIFF INFO ，在 windows 乱码，我就试图用 windows936 存了一下。

这样做的结果是， windows 文管确实可以正常读 RIFF INFO 了，但是我的 mp3 还是读不了，就比较魔幻。
